src/hooks/useFoxyMarketData/useFoxyMarketData.tsx

```diff
@@ -0,0 +1,32 @@
+import { foxyAddresses } from '@shapeshiftoss/investor-foxy'
+import { useFoxy } from 'features/defi/contexts/FoxyProvider/FoxyProvider'
+import { useCallback, useEffect, useState } from 'react'
+import { selectAssetById } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+type UseFoxyMarketDataReturnType = {
+  maxTotalSupply?: string
+}
+
+// TODO use constant from constants file when get FOX wire up is merged
+const FoxyAssetId = 'eip155:1/erc20:0xdc49108ce5c57bc3408c3a5e95f3d864ec386ed3'
+
+export const useFoxyMarketData = (): UseFoxyMarketDataReturnType => {
```

Comment:
why is this not in the plugins folder, and why does this exist at all? why doesn't the new optional fields on the market service manage this stuff?

Reply:
My mistake, I should have put it into the plugin folder.
And for the existence of this code, maxSupply / totalSupply are not available for FOXy the way we are fetching it in market service, hence the direct call to investor-foxy `totalSupply ` method.

Reply:
@chxash I had another look at this, and as market-service doesn't handle FOXy (https://github.com/shapeshift/lib/pull/670), we do currently need to fetch FOXy market data for the additional fields with investor-foxy as you've noted. With that being said, I don't believe there should be any component-specific logic to fetch it, else we're going to have to do a separate call every time we need this additional market data, which is precisely what we want to avoid by making components not fire requests.

Just being in a FOXy route should be enough for these additional "resolvers" to kick in, whether that's done on web or asset-service.

Since this is very much isolated, including the hook that I've moved into plugins folder, I'm fine with it for now, but as a follow-up, we should probably fetch FOXy market data from market-service to ease the logic on the client, otherwise it's always going to be messy shoehorning logic specifically for FOXy. @0xdef1cafe WDYT?

Reply:
Actually, the current behavior is inconsistent, and the fact that FOXy is isolated (market-data fetched in Fox Page only) while Fox market data is fetched regularly from market-service and we are reactive on it across the app, is precisely the reactivity issues we want to avoid by making components not fire requests.

Keeping this one in draft for now as I intend a way to use the regular market-data web logic for it, despite it not coming from market-service.

Reply:
Removed this hook and implemented specific `maxSupply` retrieval for FOXy in `marketDataSlice.ts`.
Also removed `fallBackMaxSupply` in the process